caravel/views.py

```diff
@@ -1588,19 +1588,53 @@ def activity_per_day(self):
 
     @api
     @has_access_api
-    @expose("/tables/<db_id>/<schema>")
+    @expose("/schemas/<db_id>")
+    @cached_view(timeout=600)
+    def schemas(self, db_id):
+        # db_id = request.args.get('db_id')
+        database = (
+            db.session
+            .query(models.Database)
+            .filter_by(id=db_id)
+            .one()
+        )
+        return Response(
+            json.dumps({'schemas': database.all_schema_names()}),
+            mimetype="application/json")
+
+    @api
+    @has_access_api
+    @expose("/tables/<db_id>/<schema>/")
+    @cached_view(timeout=600)
```

Comment:
If user creates schema / table - he already knows the name and it shouldn't be a big deal.
I can run some research to take a look often are schema's created if you are interested, I would expect it to be less than 1 a day.